packages/presentation/lib/screen/main_my_bloc.dart

```diff
@@ -0,0 +1,94 @@
+import 'package:domain/usecase/game_state.dart';
+import 'package:domain/usecase/generate_number_usecase.dart';
+import 'package:domain/usecase/check_state_usecase.dart';
+import 'package:flutter/material.dart';
+import 'package:presentation/base/bloc.dart';
+import '../base/bloc_dialog.dart';
+import 'main_data.dart';
+
+abstract class MainBloc extends Bloc {
+  factory MainBloc(
+    CheckStateUseCase checkStateUseCase,
+    GenerateNumberUseCase generateNumberUseCase,
+  ) =>
+      MainBlocImpl(checkStateUseCase, generateNumberUseCase);
+
+  TextEditingController get editController;
+
+  void generateRandomNumber();
+}
+
+class MainBlocImpl extends BlocImpl implements MainBloc {
+  int randomNumber = 1;
+  int number = 0;
+  int counter = 0;
+  final _editController = TextEditingController();
+  final _screenData = MainData.init();
+  final CheckStateUseCase _checkStateUseCase;
+  final GenerateNumberUseCase _generateNumberUseCase;
+
+  @override
+  TextEditingController get editController => _editController;
+
+  MainBlocImpl(
+    this._checkStateUseCase,
+    this._generateNumberUseCase,
+  );
+
+  @override
+  void initState() {
+    super.initState();
+    _updateData(data: _screenData);
+  }
+
+  _updateData({MainData? data}) {
+    handleData(
+      data: data,
+    );
+  }
+
+  _showAlert(String message) {
+    showAlert(event: GameDialog(message));
+  }
+
+  @override
+  void generateRandomNumber() {
+    _screenData.randomNumber = _generateNumberUseCase();
+    _screenData.isDisabled = false;
+    _updateData(
+      data: _screenData,
+    );
+  }
+
+  void checkNumber() {
+    number = int.tryParse(_editController.text)!;
```

Comment:
Don't know how to use tryParse then. Because my number is int, but tryParse expects int? value.

Reply:
1) u can use parse. it will return non-nullable value. but the problem is that if value cannot be parsed it will throw exception
2) work with number as with nullable field. it's of
3) set default value for null case